File: src/_03_BehavioralPattern/_03_04_Iterator/java/before/Board.java
package _03_BehavioralPattern._03_04_Iterator.java.before;

import java.util.ArrayList;
import java.util.List;

public class Board {
  
  List<Post> posts = new ArrayList<>();

  public List<Post> getPosts() {
    return this.posts;
  }
  public void setPosts(List<Post> posts) {
    this.posts = posts;
  }

  public void addPost(String content) {
    this.posts.add(new Post(content));
  }
  
}

File: src/myTest/Iterator/Board.js
import Post from "./Post.js";

export default class Board {
  /** @type { Post[] } */
  #posts;

  constructor() {
    this.#posts = []
  }

  /** @returns { Post[] } */
  getPosts() {
    return this.#posts;
  }
  /** @param { Post[] } posts */
  setPosts(posts) {
    this.#posts = posts;
  }

  /** @param { string } content */
  addPost(content) {
    this.#posts.push(new Post(content));
  }
}
